### PR TITLE
[cli] add command to get tcp results

### PR DESCRIPTION
**Pull request by Irving-cl**
This PR updates the cli command of TCP benchmark so that the result
of TCP benchmark could be queried any time. This enables test scripts
to automate TCP benchmark test. Currently the command `tcp benchmark`
is asynchronous and return immediately. If we use `ot-ctl tcp benchmark`
on posix, it's hard to fetch the test results. With the newly added
command, we can write a script to poll and get the result.

### CODE REVIEW

File: src/cli/cli_tcp.cpp [hunk 1]
```diff
@@ -377,49 +378,73 @@ template <> otError TcpExample::Process<Cmd("benchmark")>(Arg aArgs[])
 {
     otError error = OT_ERROR_NONE;
 
-    VerifyOrExit(!mSendBusy, error = OT_ERROR_BUSY);
-    VerifyOrExit(mBenchmarkBytesTotal == 0, error = OT_ERROR_BUSY);
-
-    if (aArgs[0].IsEmpty())
+    if (aArgs[0] == "result")
     {
-        mBenchmarkBytesTotal = OPENTHREAD_CONFIG_CLI_TCP_DEFAULT_BENCHMARK_SIZE;
```
Comment (abtink) [inline src/cli/cli_tcp.cpp:385]: We can have a common `OutputFormat("TCP Benchmakr Status: ")` at the top and then shorten the `OutputLine()` below to show the status?

**Review by abtink — APPROVED** [review]
LGTM. Thanks. 👍

**Review by samkumar — APPROVED** [review]
This looks good to me.

One minor thing to consider (which I will leave up to your discretion): When I see a command like `tcp benchmark start`, I expect there to be an analogous command like `tcp benchmark stop` or `tcp benchmark end`. Given that no such analogous command exists, you may consider renaming the command to `tcp benchmark run`. That said, I don't feel strongly about this, and am OK with merging regardless.

**Comment by jwhui** [comment]
> When I see a command like tcp benchmark start, I expect there to be an analogous command like tcp benchmark stop or tcp benchmark end. Given that no such analogous command exists, you may consider renaming the command to tcp benchmark run.

I agree. We should either have a matching `stop` command or change to `run`.

**Pull request merged**